src/main/java/uk/gov/hmcts/darts/audio/service/impl/InboundUnstructuredDataStoreStoreDeleterImpl.java

```diff
@@ -0,0 +1,117 @@
+                                                        ExternalObjectDirectoryRepository externalObjectDirectoryRepository,
+                                                        ObjectDirectoryStatusRepository objectDirectoryStatusRepository,
+                                                        ExternalLocationTypeRepository externalLocationTypeRepository,
+                                                        DataManagementConfiguration dataManagementConfiguration, UserAccountRepository userAccountRepository) {
+        this.dataManagementDao = dataManagementDao;
+        this.externalObjectDirectoryRepository = externalObjectDirectoryRepository;
+        this.objectDirectoryStatusRepository = objectDirectoryStatusRepository;
+        this.externalLocationTypeRepository = externalLocationTypeRepository;
+        this.dataManagementConfiguration = dataManagementConfiguration;
+        this.userAccountRepository = userAccountRepository;
+    }
+
+    @Transactional
+    public List<ExternalObjectDirectoryEntity> delete() {
+
+        Optional<UserAccountEntity> systemUser = getSystemUser();
+
+        if (systemUser.isEmpty()) {
+            throw new DartsApiException(ExternalDataDeleterException.MISSING_SYSTEM_USER);
+        }
+
+        ObjectDirectoryStatusEntity markedForDeletionStatus = getMarkedForDeletionStatus();
+        List<ExternalObjectDirectoryEntity> inboundData = deleteExternalData(
+            externalLocationTypeRepository.getReferenceById(INBOUND.getId()), markedForDeletionStatus,
+            dataManagementConfiguration.getInboundContainerName(),
+            systemUser.get()
+        );
+
+        List<ExternalObjectDirectoryEntity> unstructuredData = deleteExternalData(
+            externalLocationTypeRepository.getReferenceById(UNSTRUCTURED.getId()),
+            markedForDeletionStatus,
+            dataManagementConfiguration.getUnstructuredContainerName(),
+            systemUser.get()
+        );
+
+        inboundData.addAll(unstructuredData);
+
+        return inboundData;
+
+    }
+
+    private List<ExternalObjectDirectoryEntity> deleteExternalData(ExternalLocationTypeEntity externalLocationType,
+                                                                   ObjectDirectoryStatusEntity markedForDeletionStatus,
+                                                                   String containerName, UserAccountEntity systemUser) {
+        List<ExternalObjectDirectoryEntity> externalData = externalObjectDirectoryRepository.findByExternalLocationTypeAndMarkedForDeletion(
+            externalLocationType,
+            markedForDeletionStatus
+        );
+
+        BlobContainerClient containerClient = dataManagementDao.getBlobContainerClient(containerName);
+
+        ObjectDirectoryStatusEntity deletedStatus = getDeletedStatus();
+        for (ExternalObjectDirectoryEntity data : externalData) {
+            dataManagementDao.getBlobClient(containerClient, data.getExternalLocation()).delete();
```

Comment:
What happens if the delete fails?
we would probably want it to log an error and continue with the rest if it can't find the file, but if it is a more serious error, e.g. cannot connect to azure, then should probably exit.

Reply:
Looking at the api there doesn't seem to be a way to know if it failed for any reason other than If the container exists.

Reply:
Found a method that will return a http response for deleting blob storage we can use that to determine if it succeeded.